quarkchain/cluster/shard.py

```diff
@@ -435,14 +435,18 @@ def __validate_block_headers(self, block_header_list: List[MinorBlockHeader]):
 class Synchronizer:
     """ Buffer the headers received from peer and sync one by one """
 
-    def __init__(self):
+    def __init__(self, instance):
```

Comment:
this will work but I'm not a fan of those circular references between instances. how about only pass in `shard.state.subscription_manager.notify` function?